plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/job/DeployImageJob.java

```diff
@@ -124,18 +124,26 @@ protected IStatus doRun(IProgressMonitor monitor) {
 		return Status.OK_STATUS;
 	}
 
-	private boolean isUpdate(Connection connection, String project, String name) {
+	private boolean isUpdateThenDo(Connection connection, String project, String name) {
 		try {
 			IDeploymentConfig dc = connection.getResource(ResourceKind.DEPLOYMENT_CONFIG, project, name);
 			IDeploymentImageChangeTrigger trigger = (IDeploymentImageChangeTrigger) dc.getTriggers().stream()
 					.filter(t->DeploymentTriggerType.IMAGE_CHANGE.equals(t.getType())).findFirst().orElse(null);
-			return trigger != null && 
-				ResourceKind.IMAGE_STREAM_TAG.equals(trigger.getKind()) && 
-				StringUtils.isNotBlank(trigger.getNamespace()) &&  
-				connection.getResource(ResourceKind.IMAGE_STREAM, trigger.getNamespace(), trigger.getFrom().getName()) != null;
-		}catch(NotFoundException e) {
+			if (trigger == null || 
+				!ResourceKind.IMAGE_STREAM_TAG.equals(trigger.getKind()) || 
+				StringUtils.isBlank(trigger.getNamespace()) ||  
+				connection.getResource(ResourceKind.IMAGE_STREAM, trigger.getNamespace(), trigger.getFrom().getName()) == null) {
+				return false;
```

Comment:
could we please have unit-tests to assert these rather complex alternative conditions?